src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,685 @@
+============== file: Google/Example/Library/V1/ResourceNames.cs ==============
+// Copyright 2016, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated code. DO NOT EDIT!
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Example.Library.V1
+{
+    public interface IResourceName{}// TODO: This will be in GAX
+    public interface ISingleResourceName:IResourceName{} // TODO: Is this required?
+
+    /// <summary>
+    /// Resource name for a shelf.
+    /// </summary>
+    public sealed partial class ShelfName : ISingleResourceName
+    {
+        private static readonly PathTemplate s_template = new PathTemplate("shelves/{shelf_id}");
+
+        /// <summary>
+        /// Parses a shelf resource name in string form into a <see cref="ShelfName"/>.
+        /// </summary>
+        /// <param name="shelfName">The shelf resource name in string form. Must not be <c>null</c>.</param>
+        public static ShelfName Parse(string shelfName)
+        {
+            GaxPreconditions.CheckNotNull(shelfName, nameof(shelfName));
+            ResourceName resourceName = s_template.ParseName(shelfName);
+            return new ShelfName(resourceName[0]);
+        }
+
+        public static bool TryParse(string shelfName, out ShelfName result)
+        {
+            GaxPreconditions.CheckNotNull(shelfName, nameof(shelfName));
+            ResourceName resourceName;
+            if (s_template.TryParseName(shelfName, out resourceName))
+            {
+                result = new ShelfName(resourceName[0]);
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a shelf name from its component parts.
+        /// </summary>
+        /// <param name="shelfIdId">The shelfId ID. Must not be <c>null</c>.</param>
+        public ShelfName(string shelfIdId)
+        {
+            ShelfIdId = GaxPreconditions.CheckNotNull(shelfIdId, nameof(shelfIdId));
+        }
+
+        /// <summary>
```

Comment:
Fun option: when C# 7 is out, we may want to add `Deconstruct` methods which would allow 

```csharp
var (projectId, topicId) = topicName;
```

We can add that now, of course, but it won't be very useful yet. Worth doing eventually.